lib/Conversion/HWToSystemC/HWToSystemC.cpp

```diff
@@ -144,8 +150,43 @@ static void populateLegality(ConversionTarget &target) {
   target.addLegalOp<hw::ConstantOp>();
 }
 
-static void populateOpConversion(RewritePatternSet &patterns) {
-  patterns.add<ConvertHWModule, ConvertOutput>(patterns.getContext());
+static void populateOpConversion(RewritePatternSet &patterns,
+                                 TypeConverter &typeConverter) {
+  patterns.add<ConvertHWModule, ConvertOutput>(typeConverter,
+                                               patterns.getContext());
+}
+
+static void populateTypeConversion(TypeConverter &converter) {
+  converter.addConversion([](Type type) { return type; });
+  converter.addConversion([](IntegerType type) -> Type {
+    auto bw = type.getIntOrFloatBitWidth();
+    if (bw == 1)
+      return type;
+    if (bw <= 64 && type.isSigned())
+      return systemc::IntType::get(type.getContext(), bw);
+    if (bw <= 64)
+      return UIntType::get(type.getContext(), bw);
```

Comment:
Nitty consideration:

```suggestion
    if (bw <= 64) {
      if (type.isSigned)
        return systemc::IntType::get(type.getContext(), bw);
      return UIntType::get(type.getContext(), bw);
    }
```